File: Backend/src/main/java/com/assignment/courseManagementSystem/services/contract/InstructorService.java
package com.assignment.courseManagementSystem.services.contract;


import com.assignment.courseManagementSystem.dto.InstructorDto;
import org.springframework.data.domain.Page;

public interface InstructorService {

    InstructorDto createInstructor(InstructorDto dto);

    InstructorDto getInstructorById(Long id);

    Page<InstructorDto> getInstructors(
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    InstructorDto updateInstructor(Long id, InstructorDto dto);

    void deleteInstructor(Long id);
}

File: frontend/services/instructorService.js
import api from "./api";

export const getInstructors = (params) => {
  return api.get("/api/instructors", {
    params: {
      page: params.page,
      size: params.size,
      search: params.search,
      sortBy: params.sortBy,
      sortDir: params.sortDir,
    },
  });
};

export const createInstructor = (data) => {
  return api.post("/api/instructors", data);
};

export const updateInstructor = (id, data) => {
  return api.put(`/api/instructors/${id}`, data);
};

export const deleteInstructor = (id) => {
  return api.delete(`/api/instructors/${id}`);
};
